internal/docker/deployment.go

```diff
@@ -60,31 +65,13 @@ func (d *Deployment) RoundTripper() http.RoundTripper {
 	return &RoundTripper{Deployment: d}
 }
 
-// Client returns a CSAPI client targeting the given hsName, using the access token for the given userID.
-// Fails the test if the hsName is not found. Returns an unauthenticated client if userID is "", fails the test
-// if the userID is otherwise not found.
-func (d *Deployment) Client(t *testing.T, hsName, userID string) *client.CSAPI {
-	t.Helper()
+func (d *Deployment) Register(t *testing.T, hsName string, opts helpers.RegistrationOpts) *client.CSAPI {
```

Comment:
When it comes to writing individual tests, I'd like to be able to still do something like `RegisterNamed(t, "alice") like we do in SS. But I guess that can just be a wrapper on top of this interface?

Reply:
Yes, see next PR.